packages/payment-detection/src/any/any-to-eth-proxy-detector.ts

```diff
@@ -0,0 +1,128 @@
+}
+
+const PROXY_CONTRACT_ADDRESS_MAP: IProxyContractVersion = {
+  ['0.1.0']: '0.1.0',
+};
+
+/**
+ * Handle payment networks with ETH input data extension
+ */
+export default class AnyToEthFeeProxyDetector extends AnyToAnyDetector<PaymentTypes.IETHPaymentEventParameters> {
+  /**
+   * @param extension The advanced logic payment network extensions
+   */
+  public constructor({
+    advancedLogic,
+    currencyManager,
+  }: {
+    advancedLogic: AdvancedLogicTypes.IAdvancedLogic;
+    currencyManager: ICurrencyManager;
+  }) {
+    super(
+      advancedLogic.extensions.anyToEthProxy,
+      ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ETH_PROXY,
+      currencyManager,
+    );
+  }
+
+  /**
+   * Extracts payment events of an address matching an address and a payment reference
+   *
+   * @param address Address to check
+   * @param eventName Indicate if it is an address for payment or refund
+   * @param requestCurrency The request currency
+   * @param paymentReference The reference to identify the payment
+   * @param paymentNetwork the payment network
+   * @returns The balance
+   */
+  protected async extractEvents(
+    address: string,
+    eventName: PaymentTypes.EVENTS_NAMES,
+    requestCurrency: RequestLogicTypes.ICurrency,
+    paymentReference: string,
+    paymentNetwork: ExtensionTypes.IState<any>,
+  ): Promise<PaymentTypes.ETHPaymentNetworkEvent[]> {
+    const network = this.getPaymentChain(requestCurrency, paymentNetwork);
+
+    const conversionProxyContract = await this.safeGetProxyArtifact(
+      network,
+      paymentNetwork.version,
+    );
+
+    if (!conversionProxyContract) {
+      throw Error('ETH conversion proxy contract not found');
+    }
+
+    const currency = this.currencyManager.fromStorageCurrency(requestCurrency);
+    if (!currency) {
+      throw Error('requestCurrency not found in currency manager');
+    }
+
+    const proxyInfoRetriever = new ProxyInfoRetriever(
+      currency,
+      paymentReference,
+      conversionProxyContract.address,
+      conversionProxyContract.creationBlockNumber,
+      conversionProxyContract.abi,
+      address,
+      eventName,
+      network,
+      undefined,
+      paymentNetwork.values?.maxRateTimespan,
+    );
+
+    return await proxyInfoRetriever.getTransferEvents();
+  }
+
+  /**
+   * Get the network of the payment
+   *
+   * @param requestCurrency The request currency
+   * @param paymentNetwork the payment network
+   * @returns The network of payment
+   */
+  protected getPaymentChain(
+    _requestCurrency: RequestLogicTypes.ICurrency,
+    paymentNetwork: ExtensionTypes.IState<any>,
+  ): string {
+    const network = paymentNetwork.values.network;
+    if (!network) {
+      throw Error('paymentNetwork.values.network must be defined');
+    }
+    return network;
+  }
+
+  /*
+   * Fetches events from the Ethereum Proxy, or returns null
+   */
+  private async safeGetProxyArtifact(network: string, paymentNetworkVersion: string) {
```

Comment:
this is no longer necessary, we have `getOptionalDeploymentInformation` that doesn't throw